### PR TITLE
[BEAM-759] Remove blocking dataflow runner

### PR DESCRIPTION
**Pull request by sb2nov**
R: @aaltay PTAL 

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by sb2nov** [comment]
retest this please

File: sdks/python/apache_beam/runners/dataflow_runner.py [hunk 1]
```diff
@@ -180,9 +175,6 @@ def run(self, pipeline):
     self.result = DataflowPipelineResult(
         self.dataflow_client.create_job(self.job), self)
 
-    if self.result.has_job and self.blocking:
-      self.result.wait_until_finish()
-
     return self.result
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow_runner.py]: Do we need `self.result` any more? Can we just return the `DataflowPipelineResult` object?

Comment (sb2nov) [inline sdks/python/apache_beam/runners/dataflow_runner.py]: Good point, will fix.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**